src/renderer/dx/CustomTextLayout.cpp

```diff
@@ -390,6 +390,17 @@ CATCH_RETURN()
         std::vector<DWRITE_SHAPING_TEXT_PROPERTIES> textProps(textLength);
         std::vector<DWRITE_SHAPING_GLYPH_PROPERTIES> glyphProps(maxGlyphCount);
 
+        const auto features = _fontRenderData->DefaultFontFeatures();
+        std::vector<DWRITE_FONT_FEATURE> featureVector;
+        for (const auto& [tag, param] : features)
+        {
+            featureVector.push_back(DWRITE_FONT_FEATURE{ tag, param });
+        }
+        DWRITE_FONT_FEATURE* featureList = &featureVector[0];
+        DWRITE_TYPOGRAPHIC_FEATURES typographicFeatures = { &featureList[0], gsl::narrow<uint32_t>(featureVector.size()) };
+        DWRITE_TYPOGRAPHIC_FEATURES const* typographicFeaturesPointer = &typographicFeatures;

```

Comment:
Can we move at least part of these into `DxFontRenderData`? First, repeatedly creating `featureVector` and pushing elements into it might be bad for performance. Second, I would love to utilize `DxFontRenderData`  for this layer of abstraction, which is metadata -> DWrite structures.

Reply:
That makes sense! We no longer involve DxFontInfo